base/compiler/utilities.jl

```diff
@@ -156,7 +156,15 @@ function subst_trivial_bounds(@nospecialize(atypes))
     end
     v = atypes.var
     if isconcretetype(v.ub) || v.lb === v.ub
-        return subst_trivial_bounds(atypes{v.ub})
+        subst = try
+            atypes{v.ub}
+        catch
+            # Note in rare cases a var bound might not be valid to substitute.
+            nothing
+        end
+        if subst !== nothing
+            return subst_trivial_bounds(subst)
+        end
```

Comment:
Oh boy, this is a nice example for justifying a
```julia
try
catch
else
    # code that runs when no exception was thrown
end
```
construct. But given that we don't have it, I guess this is fine.